src/projection.rs

```diff
@@ -37,10 +41,10 @@ macro_rules! unsafe_pinned {
     ($f:tt: $t:ty) => (
         #[allow(unsafe_code)]
         fn $f<'__a>(
-            self: $crate::core_reexport::pin::Pin<&'__a mut Self>
-        ) -> $crate::core_reexport::pin::Pin<&'__a mut $t> {
+            self: ::core::pin::Pin<&'__a mut Self>
+        ) -> ::core::pin::Pin<&'__a mut $t> {
             unsafe {
-                $crate::core_reexport::pin::Pin::map_unchecked_mut(
+                ::core::pin::Pin::map_unchecked_mut(
```

Comment:
I don't think these change (`$crate::core_reexport` -> `::core`) is necessary. (futures crate [prefers this pattern.](https://github.com/rust-lang-nursery/futures-rs/blob/master/futures-core/src/task/poll.rs#L7))

Reply:
I reverted the changes, I'd like to know how is it more powerful ?

Reply:
> I'd like to know how is it more powerful ?

`::core` may cause a compile error or call the wrong type if [another crate that named 'core' exists](https://github.com/search?l=Rust&q=%22extern+crate+%22%22+as+core%22&type=Code).
[`$crate`](https://doc.rust-lang.org/reference/macros-by-example.html#hygiene) refers to the defined crate, so it will work correctly even if the name of `core` or `pin-utils` is changed or overwritten name with another crate.

For example, if we make the following changes to the first commit of this PR,

```diff,rust
diff --git a/tests/stack_pin.rs b/tests/stack_pin.rs
index 4b6758d..d279908 100644
--- a/tests/stack_pin.rs
+++ b/tests/stack_pin.rs
@@ -1,7 +1,9 @@
 #![forbid(unsafe_code)] // pin_mut! is completely safe.
 
+extern crate alloc as core;
+
 use pin_utils::pin_mut;
-use core::pin::Pin;
+use std::pin::Pin;
 
 #[test]
 fn stack_pin() {
```

We will get an error.

```rust
error[E0433]: failed to resolve: could not find `pin` in `core`
  --> tests/stack_pin.rs:12:5
   |
12 |     pin_mut!(foo);
   |     ^^^^^^^^^^^^^^ could not find `pin` in `core`
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
```

Actually I think this should rarely be necessary, but [a macro in this crate is re-exported by futures](https://github.com/rust-lang-nursery/futures-rs/blob/master/futures-util/src/lib.rs#L34), so I think it would be preferable to support this like futures.

Reply:
Ok thank you for the explaination